### ISSUE TITLE
No autowiring for subcommands

### ISSUE THREAD

**Issue by CC007** [Alan-Gomes/mcspring-boot#7]
When executing a subcommand, any object that should be autowired is `null` during the `run()` method.

This only seems to happen with subcommands (like `headsplugin update`). Top level commands (like `headsplugin`) work as expected.

___
> Creator of HeadsPluginAPI

**Comment by Alan-Gomes** [comment]
Hi, thanks for the feedback! Can you provide some test code? I ran a test plugin (using the version 0.10.0) and the dependency injection occurred normally.

![Screenshot](https://user-images.githubusercontent.com/17713246/58027818-d0c73780-7aef-11e9-9a5e-cd3eaf6a866d.png)

**Comment by CC007** [comment]
I have my subcommands in a separate java file. Could that be the issue? Is there a way to make that work?

**Comment by CC007** [comment]
Here is the command:
```
@Component
@Command(
	name = "headspluginapi",
	versionProvider = PluginVersionProvider.class,
	description = "Provides the commands for HeadsPlugin",
	aliases = {"headsplugin", "hpa", "hp"},
	mixinStandardHelpOptions = true,
	subcommands = {UpdateCategoryCommand.class}
)
public class HeadsPluginCommand extends AbstractCommand
{
	@Override
	public void run()
	{
		CommandLine cmd = commandSpec.commandLine();
		if (cmd.isVersionHelpRequested()) {
			context.getPlayer().sendMessage(commandSpec.version());
		}
		if (cmd.isUsageHelpRequested()) {
			context.getPlayer().sendMessage(chatManager.getChatBanner() + "\n" + StringUtils.remove(cmd.getUsageMessage(), '\r') + "\n ");
		}
	}
}

```

And here the subcommand
```
@Component
@Command(
	name = "update",
	versionProvider = PluginVersionProvider.class,
	mixinStandardHelpOptions = true,
	description = "Update all or a specific category",
	aliases = {"up", "updateheads", "uh","uphds"}

)
public class UpdateCategoryCommand extends AbstractCommand
{
	@Parameters(
		index = "0",
		defaultValue = "all",
		//completionCandidates = ..., TODO complete from category names
		description = "The category to update. If this parameter is not specified, all categories will be updated",
		paramLabel = "categoryName"
	)
	private String categoryName;

	@Override
	public void run()
	{

		if (context.getPlayer() != null) {
			context.getPlayer().sendMessage(chatManager.getChatPrefix() + "This might actually do stuff to " + categoryName + " in the future");
		}
		else if(context.getSender() instanceof ConsoleCommandSender) {
			context.getSender().sendMessage(chatManager.getConsolePrefix() + "This might actually do stuff to " + categoryName + " in the future");
		}
	}
}
```

**Comment by Alan-Gomes** [comment]
Yes, this feature can be supported, I just created a commit to fix the injection problems on annotated subcommands, you can check how it works on [this test](https://github.com/Alan-Gomes/mcspring-boot/commit/d913a98ecab666892c11cc5273f0c5b96acc91ab#diff-8e2f5975e290b3a53ddc4dabef548a13R117). This change will be available on 0.11.0

Thank you!
